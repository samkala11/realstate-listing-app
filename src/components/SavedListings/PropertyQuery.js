import React, {useEffect, useState} from 'react';
import Property from '../Property/Property';
import Paper from '@material-ui/core/Paper';
import Skeleton from '@material-ui/lab/Skeleton';


import { getListing } from '../../utils/listings_api_util';

const PropertyQuery = ({mlsId, docId, deleteListing}) => {

    const [listingInfo, setListingInfo] = useState(null);

    window.listingInfoStateQuery = listingInfo;

    useEffect(() => {
        getListing(mlsId)
        .then(res => {
            console.log('async get 1 listing');
            console.log(res);
            setListingInfo(res);
        })

    }, [mlsId])
    

    return(
        <>

            {listingInfo ? 
            <Paper data-testid="property-query-container" id="property-container" elevation={3}>
                <Property
                    key = { listingInfo.mlsId }
                    mlsId = {listingInfo.mlsId } 
                    photosArray = {listingInfo ? listingInfo.photos : []}
                    price = {listingInfo.listPrice}
                    bedrooms = { listingInfo && listingInfo.property ? listingInfo.property.bedrooms : null}
                    bathsFull = { listingInfo && listingInfo.property ? listingInfo.property.bathsFull : null}
                    bathsHalf = {  listingInfo && listingInfo.property ? listingInfo.property.bathsHalf : null}
                    stories = {listingInfo && listingInfo.property ? listingInfo.property.stories : null}
                    listingDate = {listingInfo.listDate}
                    saveProperty = {deleteListing}
                    saved ={true}
                    id ={docId}
                    
                />  
            </Paper> :
            <div data-testid="property-query-container">   
                <Skeleton variant="rect" width={350} height={280} />
                <Skeleton variant="text" />
                <Skeleton variant="text" />
                <Skeleton variant="text" />
            </div>
}


        </>
    )
}

export default PropertyQuery;