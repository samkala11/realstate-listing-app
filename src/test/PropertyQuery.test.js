import { render, screen, cleanup } from '@testing-library/react';
import PropertyQuery from '../components/SavedListings/PropertyQuery';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('Renders PropertyQuery page successfully', () => {

    render(<PropertyQuery  
            mlsId={5990355}
        />);

    const propertyQueryContainer = screen.getByTestId('property-query-container');
    expect(propertyQueryContainer).toBeInTheDocument();
});

test('matches PropertyQuery page snapshot with given props', () => {
    const tree = renderer.create(<PropertyQuery 
            mlsId={5990355}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})