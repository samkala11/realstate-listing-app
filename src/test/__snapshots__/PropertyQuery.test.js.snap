// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches PropertyQuery page snapshot with given props 1`] = `
<div
  data-testid="property-query-container"
>
  <span
    className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
    style={
      Object {
        "height": 280,
        "width": 350,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
</div>
`;
